src/App.js

```diff
@@ -21,45 +21,132 @@ const generateSquares = () => {
       currentId += 1;
     }
   }
-
   return squares;
 };
 
 const App = () => {
-  // This starts state off as a 2D array of JS objects with
-  // empty value and unique ids.
+
   const [squares, setSquares] = useState(generateSquares());
+  const [player, setPlayer] = useState(PLAYER_1);
+  const [checkWinner, setCheckWinner] = useState(false);
+  const [winner, setWinner] = useState('');
+
+  const updateSquare = (id) => {
+
+    // stop changing state
+    if (winner) {
+      return;
+    }
+
+    const sqs = squares.map((rows) =>
+      rows.map((singleSquare) => {
+        if (singleSquare.id === id) {
+          if (player === PLAYER_1 && singleSquare.value === '') {
+            singleSquare.value = 'x';
+          } else if (player === PLAYER_2 && singleSquare.value === '') {
+            singleSquare.value = 'o';
+          }
+        }
+
+        return singleSquare;
+      })
+    );
+
+    setPlayer(player === PLAYER_1 ? PLAYER_2 : PLAYER_1);
+    setSquares(sqs);
+    setWinner(pickWinner(checkForWinner()));
+  };
+
+
+
+
+// const handleSquareClick = (id) => {
+//   const newSquares = [...squares];
+//   for(let row of newSquares) {
+//     for(let col of row) {
+//       if (col.id === id) {
+//         col.value = player;
+//       }
+//     }
+//   }
+//   if (player === player1) {
+//     setPlayer(player2);
+//   } else {setPlayer(player1);}
+//   setSquares(newSquares);
+
+// }
```

Comment:
let's remove any unused code! if you want to keep it for reference, think about creating a file that holds notes or pieces of code!

```suggestion
```